### PR TITLE
Use typescript AST

### PR DESCRIPTION
**Pull request by vincaslt**
All tests passing, and that's about it. I haven't tested if it fixed any issues (like #155 or #166 ), or in real-world projects.

This includes the changes from #153 because I don't think it makes sense do this work twice. It should be possible to merge without conflicts were both these PRs got merged.

This PR rewrites lexers to use typescript AST, so typescript is supported implicitly. I think jsx and javascript lexers could be joined somehow to eliminate the distinction, but not sure.

I would say, this is basically a starting point for any effort to improve this tool. As a first step, by dropping acorn parser, because it needed several other dependencies, some of which are incompatible with each other, and causes problems when upgrading.

### CODE REVIEW

**Comment by karellm** [comment]
This branch now needs a rebase :)

**Comment by vincaslt** [comment]
Rebased

File: test/parser.test.js [hunk 1]
```diff
@@ -747,7 +747,7 @@ describe('parser', () => {
         first: '',
         second: '',
         third: {
-          first: 'Hello <1>{{name}}</1>, you have {{count}} unread message. <5>Go to messages</5>.',
+          first: 'Hello <strong>{{name}}</strong>, you have {{count}} unread message. <5>Go to messages</5>.',
```
Comment (vincaslt) [inline test/parser.test.js]: I believe this is a mistake, it shouldn't parse a basic tag here, will try to fix this.

Comment (vincaslt) [reply test/parser.test.js]: Fixed

**Comment by karellm** [comment]
I just reviewed the PR, it looks really great. Thanks a lot for putting the work to removing `acorn`.

I would like to make you a collaborator of this project. It doesn't really require more of you than what you are already doing. The only thing I'd ask is that we keep working with PRs with peer reviews and that we don't publish directly to `master`. Let me know if that interests you :)

**Comment by vincaslt** [comment]
Yeah, sure

**Pull request merged**